outline/client.go

```diff
@@ -25,3 +29,28 @@ type Client struct {
 	transport.StreamDialer
 	transport.PacketListener
 }
+
+// NewShadowsocksClientFromJSON creates a new Shadowsocks client from a JSON
+// formatted configuration.
+func NewShadowsocksClientFromJSON(configJSON string) (*Client, error) {
+	config, err := parseConfigFromJSON(configJSON)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse Shadowsocks configuration JSON: %w", err)
+	}
+
+	var prefixBytes []byte = nil
+	if len(config.Prefix) > 0 {
+		if p, err := utf8.DecodeUTF8CodepointsToRawBytes(config.Prefix); err != nil {
+			return nil, fmt.Errorf("failed to parse prefix string: %w", err)
+		} else {
+			prefixBytes = p
+		}
+	}
+
+	sd, pl, err := internal.NewShadowsocksConn(config.Host, int(config.Port), config.Method, config.Password, prefixBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{*sd, *pl}, nil
```

Comment:
It will introduce cyclic dependency in the old architecture. Also I would try to avoid any more dependencies to the deprecated `shadowsocks` package (rather I would depend on the `internal/shadowsocks` package), so in the future we can remove legacy packages.